templates/federated_reporting/psql_wrapper.sh

```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+
+# run an arbitrary psql command as cfpostgres user
+# expectation is that a line will be output with exit_code=<n>
+# this exit code <n> will then be used to exit
+# This design enables cfengine policy to run a commands promise
+# that can be kept, repaired or failed depending on an exit
+# code from psql since psql can't get different codes from scripts.
+
+if [ $# -ne 2 ]; then
+    echo "Usage: $0 [DBNAME] [SQL STRING]"
+    exit 2
+fi
+
+TMP=$(mktemp)
+TMP2=$(mktemp)
+OUT=$(su - cfpostgres --command "psql --quiet --tuples-only --no-align --no-psqlrc \"$1\" --command=\"$2\"" 2> $TMP)
+RETURN_CODE=$?
+ERR=$(<$TMP)
+
+EXIT_CODE=$(echo $TMP | awk -F= '{ if ( /exit_code/ ) print $2}')
+
+rm $TMP $TMP2
+
+if [ $RETURN_CODE -ne 0 ]; then
+    echo "$OUT"
+    echo "$ERR"
+    exit 2; # failed
```

Comment:
Shouldn't this just let it go further to the `exit $EXIT_CODE` below?

Reply:
I should add a comment about what exit codes mean from this script and how they are used from policy.

psql return codes are not sufficient to return to policy. Returning 2 for any psql failure is the correct thing to do.

psql:
```
EXIT STATUS
       psql returns 0 to the shell if it finished normally, 1 if a fatal error of its own
       occurs (e.g. out of memory, file not found), 2 if the connection to the server went
       bad and the session was not interactive, and 3 if an error occurred in a script and
       the variable ON_ERROR_STOP was set.
```

But I need to always return exit codes as follows:
```
body classes superhub_schema_exit_codes
{
  kept_returncodes => { "0" };
  repaired_returncodes => { "1" };
  failed_returncodes => { "2" };
}
```